go/chat/push.go

```diff
@@ -311,6 +313,45 @@ func (g *PushHandler) TlfResolve(ctx context.Context, m gregor.OutOfBandMessage)
 	return nil
 }
 
+// squashChanMention will silence a chanMention if sent by a non-admin on a
+// large (> chat1.MaxChanMentionConvSize member) team. The server drives this
+// for mobile push notifications, client checks this for desktop notifications
+func (g *PushHandler) squashChanMention(ctx context.Context, conv *chat1.ConversationLocal,
+	mvalid chat1.MessageUnboxedValid) (bool, error) {
+	// Verify the chanMention is for a TEAM that is larger than
+	// MaxChanMentionConvSize
+	if conv == nil ||
+		conv.Info.MembersType != chat1.ConversationMembersType_TEAM ||
+		len(conv.Info.Participants) < chat1.MaxChanMentionConvSize {
+		return false, nil
+	}
+	teamID, err := keybase1.TeamIDFromString(mvalid.ClientHeader.Conv.Tlfid.String())
+	if err != nil {
+		return false, err
+	}
+	extG := g.G().ExternalG()
+	team, err := teams.Load(ctx, extG, keybase1.LoadTeamArg{
```

Comment:
I wonder if all this code should just be in a helper function in the `teams` package.

Reply:
most of it is in the team package, unless we made a chat specific helper (took a `tlfid` string in) we'd only save the upak load to the get sender's UV, I don't think it's worth it